File: cpp/1324.cpp
#define max(x, y) ((x > y) ? x : y)

class Solution {
public:
    vector<string> printVertically(string& w) {
        int c = 1, m = 0, n = w.length(), p = 0;
        for (int i = 0; i < n; i++)
            if (w[i] == ' ') {
                m = max(m, i - p);
                p = i + 1;
                c++;
            }
        m = max(n - p, m);
        char** r = (char**)malloc(sizeof(char*) * m);
        for (int i = 0; i < m; i++)
            r[i] = (char*)malloc(sizeof(char) * c);
        int i = 0;
        for (int j = 0; j < c; j++) {
            for (int k = 0; k < m; k++) {
                if (i < n && w[i] != ' ')
                    r[k][j] = w[i++];
                else
                    r[k][j] = ' ';
            }
            i++;
        }
        vector<string> t;
        for (int j = 0; j < m; j++) {
            int x = c - 1;
            while (x >= 0 && r[j][x] == ' ')
                x--;
            string y = "";
            while (x >= 0)
                y = r[j][x--] + y;
            t.push_back(y);
        }
        return t;
    }
};

File: python3/1324.py
class Solution:
    def printVertically(self, w: str) -> List[str]:
        c = 1
        m = 0
        n = len(w)
        p = 0
        for i in range(n):
            if w[i] == " ":
                m = max(m, i - p)
                p = i + 1
                c += 1
        m = max(n - p, m)
        r = [[0] * c for _ in range(m)]
        i = 0
        for j in range(c):
            for k in range(m):
                if i < n and w[i] != " ":
                    r[k][j] = w[i]
                    i += 1
                else:
                    r[k][j] = " "
            i += 1
        t = []
        for u in r:
            x = c - 1
            while x >= 0 and u[x] == " ":
                x -= 1
            t.append("".join(u[: (x + 1)]))
        return t
